simple_ado/wiki.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""ADO Wiki API wrapper."""
+
+import logging
+
+from simple_ado.base_client import ADOBaseClient
+from simple_ado.http_client import ADOHTTPClient, ADOResponse
+
+
+class ADOWikiClient(ADOBaseClient):
+    """Wrapper class around the ADO Wiki APIs.
+
+    :param http_client: The HTTP client to use for the client
+    :param log: The logger to use
+    """
+
+    def __init__(self, http_client: ADOHTTPClient, log: logging.Logger) -> None:
+        super().__init__(http_client, log.getChild("wiki"))
+
+    def get_page_version(self, page_id: str, wiki_id: str, project_id: str) -> ADOResponse:
+        """Get's the current version of a wiki page. This returns a required parameter for updating a wiki page.
+
+        https://docs.microsoft.com/en-us/rest/api/azure/devops/wiki/pages/get%20page%20by%20id?view=azure-devops-rest-6.1
+
+        :param page_id: Wiki page ID
+        :param wiki_id: Wiki ID or Wiki name
+        :param str project_id: The ID of the project
+
+        :returns: The ADO response with the data in it
+        """
+
+        self.log.debug(f"Get wiki page: {page_id}")
+        request_url = (
+            self.http_client.api_endpoint(is_default_collection=False, project_id=project_id)
+            + f"/wiki/wikis/{wiki_id}/pages/{page_id}?api-version=6.1-preview.1"
+        )
+        response = self.http_client.get(request_url)
+        self.http_client.validate_response(response)
+        return response.headers["ETag"]
+
+    def update_page(
+        self, page_id: str, wiki_id: str, project_id: str, content: str, current_version_etag: str
+    ) -> ADOResponse:
+        """Update a the contents of a wiki page.
+
+        https://docs.microsoft.com/en-us/rest/api/azure/devops/wiki/pages/update?view=azure-devops-rest-6.1
+
+        :param page_id: Wiki page ID
+        :param wiki_id: Wiki ID or Wiki name
+        :param str project_id: The ID of the project
+        :param str content: Content of the wiki page.
+        :param str current_version_etag: The ETag of the current wiki page to verify the update.
+
+        :returns: The ADO response with the data in it
+        """
+
+        self.log.debug(f"Updating wiki page: {page_id}")
+        request_url = (
+            self.http_client.api_endpoint(is_default_collection=False, project_id=project_id)
+            + f"/wiki/wikis/{wiki_id}/pages/{page_id}?api-version=6.1-preview.1"
+        )
+        response = self.http_client.patch(
+            request_url,
+            json_data={"content": content,},
+            additional_headers={
+                "Content-Type": "application/json",
```

Comment:
This API seems so different to the others. Data in the ETag, not setting `+patch` on the content type, etc. We'll have to keep an eye on this one as I won't be surprised if it changes rapidly.

Reply:
Yeah, I think it may have to do with the fact that the wiki endpoints only became available in recent API versions vs the rest of them. I'll definitely keep an eye on it.

I see the benefit with the ETag if you want to ensure you're updating the most up-to-date page, but it is odd that it's always required for any update.